pdns/dnsdist-lua-actions.cc

```diff
@@ -806,46 +797,68 @@ class LogAction : public DNSAction, public boost::noncopyable
     }
     return "log";
   }
+
+  void reload() override
+  {
+    reopenLogFile();
+  }
+
 private:
+  void reopenLogFile()
+  {
+    std::shared_ptr<FILE> fp = nullptr;
+
+    if (d_append) {
+      fp = std::shared_ptr<FILE>(fopen(d_fname.c_str(), "a+"), fclose);
+    }
+    else {
+      fp = std::shared_ptr<FILE>(fopen(d_fname.c_str(), "w"), fclose);
+    }
+
+    if (!fp) {
```

Comment:
would it make sense to log that we were unable to re-open the file?

Reply:
Yes! I have just done that :+1: